### PR TITLE
Exposing static method to retrieve DLEQParams.

### PR DESCRIPTION
**Pull request by armfazh**
Changes:
 - exposes a method to get DLEQParams
 - deserialization uses suiteID
 - bump up dependencies.

### CODE REVIEW

File: package.json
Comment (thibmeu) [inline package.json:1]: none of the update here seems to be related to change being made. would be good to have it in a separate commit

File: src/noble/group.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 import { bytesToNumberBE, bytesToNumberLE } from '@noble/curves/abstract/utils'
 
-import { type Deserializer,type  Group,type  GroupID, Groups } from '../groupTypes.js'
+import { type Deserializer, type Group, type GroupID, Groups } from '../groupTypes.js'
```
Comment (thibmeu) [inline src/noble/group.ts:8]: any idea why this did not get caught at lint time by the CI test suite?

File: src/oprf.ts [hunk 2]
```diff
@@ -12,7 +12,6 @@ import {
     type GroupID,
     type Scalar
 } from './groupTypes.js'
-
```
Comment (thibmeu) [inline src/oprf.ts:15]: lint enforcement seems to be insufficient in CI

File: src/oprf.ts [hunk 3]
```diff
@@ -98,14 +97,20 @@ export abstract class Oprf {
...
     }
 
     static getOprfSize(suite: SuiteID): number {
         return getOprfParams(suite)[3]
     }
 
+    static getDLEQParams(suite: SuiteID): DLEQParams {
+        const [, gid, hashID] = getOprfParams(suite)
+        const gg = Oprf.Group.fromID(gid)
+        return { gg, hashID, hash: Oprf.Crypto.hash, dst: '' }
```
Comment (thibmeu) [inline src/oprf.ts:111]: defined empty `dst` as a constant

File: src/oprf.ts [hunk 4]
```diff
@@ -129,6 +134,10 @@ export abstract class Oprf {
         this.mode = Oprf.validateMode(mode)
     }
 
+    protected getDLEQParams(): DLEQParams {
+        return { gg: this.gg, hashID: this.hashID, hash: Oprf.Crypto.hash, dst: '' }
```
Comment (thibmeu) [inline src/oprf.ts:138]:
```suggestion
        return Oprf.getDLEQParams(this.ID)
```

File: src/oprf.ts [hunk 5]
```diff
@@ -193,7 +202,8 @@ export class Evaluation {
         return res
     }
 
-    static deserialize(params: DLEQParams, bytes: Uint8Array): Evaluation {
+    static deserialize(suite: SuiteID, bytes: Uint8Array): Evaluation {
+        const params = Oprf.getDLEQParams(suite)
```
Comment (thibmeu) [inline src/oprf.ts:206]:
```suggestion
        const params = suite.getDLEQParams()
```

File: src/oprf.ts (hunk 4 above)
Comment (thibmeu) [reply src/oprf.ts:138]: other option is to not declare this method and do the same as `getGroup`, which always refers to the static method

File: src/noble/group.ts (hunk 1 above)
Comment (armfazh) [reply src/noble/group.ts:8]: this is a relatively new feature, likely there is no rule that catches this.

File: src/oprf.ts (hunk 2 above)
Comment (armfazh) [reply src/oprf.ts:15]: linter does not check for code formatting.

File: src/oprf.ts (hunk 5 above)
Comment (armfazh) [reply src/oprf.ts:206]: `suite` is light-weight string identifier, while Oprf is the object that charges with all the logic.

File: src/oprf.ts (hunk 3 above)
Comment (armfazh) [reply src/oprf.ts:111]: done

File: src/oprf.ts (hunk 2 above)
Comment (armfazh) [reply src/oprf.ts:15]: updated the configuration now

File: package.json
Comment (armfazh) [reply package.json:1]: split in two commits.

File: src/oprf.ts (hunk 4 above)
Comment (armfazh) [reply src/oprf.ts:138]: sure, avoids duplicated code.

**Comment by armfazh** [comment]
This last commit performs the actual checking of DLEQ proofs against test vectors. For some reason, this checking was omitted in previous code.

**Review by thibmeu — APPROVED** [review]
good to go
